### PR TITLE
Modulation of Sherlock

### PR DESCRIPTION
**Pull request by knijam**
Hello,
I came across your project and was interested to get started helping out.

Here I submitted a pull request from my fork of sherlock to segregate the application into small modules and make it a little more modular, I noticed it's a huge steer, however I think it would be a great idea.

If you do accept my pull request, may I suggest appending this commit into a separate branch.

# Changes:
- Creation of a sherlock folder
- SherlockData class 
- SherlockLog class
- SherlockService class

## SherlockData class:
Creating an instance of this class loads the data.json or any other json files as a configuration for the SherlockService class.

## SherlockLog class:
Initializing Logging activities and keeping track of activities.  The class committed only contain error, log and info methods for printing to the stdout using the color package.

## SherlockService class:
A unique SherlockService which performs asynchronous requests to HTTP and HTTPS web server using requests and grequests. 

Thank you :+1:

### CODE REVIEW

**Comment by TheYahya** [comment]
Thank you @knijam, Seems you did quite a job here, It took me a while to check this.
And for you to know, we discuss this before in #37 and there's already a PR at #106, so take a look at them and feel free to make changes in your PR if you think so.

**Comment by knijam** [comment]
That's alright :+1:  @TheYahya 
Yeah, I thought I'd take a little risk.  See what I could help with, but great I'll have a look at these #37 and #106.

**Comment by knijam** [comment]
Absolutely agree, also what's your view on an automated testing infrastructure? I suggested tox in issue #37 however any automation system may do in my opinion, there is this package called __nox__ however I cannot say much because I never used nox xD

**Comment by TheYahya** [comment]
I never used nox either, but seems pretty cool, I guess give it a shot.
Also take a look at #105

**Comment by knijam** [comment]
### Current outlook.
image

This is the current outlook output from all tests, all the tests are stubs at the moment :smile: but getting there.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+]
+
+# Packages part of setup
+_packages = [
+    "sherlock"
+]
+
+# Setup for the project
+setup(
+    name="sherlock",
```
Comment (TheYahya) [inline setup.py]: It should change to `thesherlock`, because sherlock already taken.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+from setuptools import setup
+
+# Current version of setup
+def _get_version():
+    return "%i.%i.%i" % (2, 3, 1)
```
Comment (TheYahya) [inline setup.py]: Current version is `0.2.6`

**Comment by TheYahya** [comment]
Thanks @knijam, seems ok to me. 
I don't know the need of `.sherlock` directory?!
And the last think is updating the `readme.md` and describe the usage, tests and anything needed.

**Comment by knijam** [comment]
Oh it was temporal folder '.sherlock' I believe I forgot to remove that commit.  @TheYahya Thanks for the review, I'll be making these changes asap :+1:

**Comment by sdushantha** [comment]
@TheYahya @hoadlck
Are we going to upload Sherlock to the PyPi?

**Comment by TheYahya** [comment]
Yes @sdushantha, It's a huge work and have lot's of conflicts, Do you like to spend time on this?

**Comment by sdushantha** [comment]
@TheYahya I dont think I will be able to do that because I dont have much time. Sorry :(

**Comment by hoadlck** [comment]
Much of the ideas of this have been incorporated into #590.

Thanks.

**Pull request closed without merging**